File: Lab 2/Ireni_100657302/kafka/topic.js
//acquire library
const Kafka = require("kafkajs").Kafka

run();

async function run() {

    try {
        //create an admin connection to create a topic
        //establish tcp connection

        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        })

        //admin interface
        const admin = kafka.admin();
        console.log("Connecting.....")
        await admin.connect()
        console.log("Connected!")

        //create a topic A-M
        await admin.createTopics({
            "topics": [{
                "topic": "Users",
                "numPartitions": 2
            }]
        })

        console.log("Done! Created Successfully")
        await admin.disconnect();

    } catch (excecption) {
        console.error(`Something bad happened ${ex}`);
    }

    finally {
        process.exit(0);
    }
}

File: Lab 2/Ireni_100657302/kafka-confluent-cloud/topic.py
from confluent_kafka.admin import AdminClient, NewTopic

def run():

    try:
        topic_list = []

        admin_client = AdminClient(
            {"bootstrap.servers": "localhost:9093, localhost:9094, localhost:9095"}
        )

        topic_list.append(NewTopic(
                "Users",
                num_partitions = 2,
                replication_factor = 1
            )
        )

        admin_client.create_topics(
            new_topics = topic_list,
            validate_only = False
        )

        print("Done! Created Successfully")

    except Exception as e:

        topic_created = "TopicAlreadyExistsError:"

        if topic_created in str(e):
            print("Topic Is Already Created")

        else: print("Something bad happened " + str(e))

run()
